subsys/mgmt/mcumgr/buf.c

```diff
@@ -42,48 +42,42 @@ static uint16_t
 cbor_nb_reader_get16(struct cbor_decoder_reader *d, int offset)
 {
 	struct cbor_nb_reader *cnr;
-	uint16_t val;
 
 	cnr = (struct cbor_nb_reader *) d;
 
-	if (offset < 0 || offset > cnr->nb->len - (int)sizeof(val)) {
+	if (offset < 0 || offset > cnr->nb->len - (int)sizeof(uint16_t)) {
 		return UINT16_MAX;
 	}
 
-	memcpy(&val, cnr->nb->data + offset, sizeof(val));
-	return cbor_ntohs(val);
+	return sys_be16_to_cpu(*(uint16_t *)(cnr->nb->data + offset));
```

Comment:
`sys_be16_to_cpu` assumes aligned memory access. I think that `sys_get_be16` should be used instead, so that potentially unaligned memory access is handled correctly.

Reply:
You are right. Thanks!